miniapp/brunel/brunel_miniapp.cpp

```diff
@@ -67,35 +63,45 @@ std::vector<int> sample_subset(int gid, int start, int end, int m) {
 class brunel_recipe: public recipe {
 public:
     brunel_recipe(cell_size_type nexc, cell_size_type ninh, cell_size_type next, double in_degree_prop,
-                  float weight, float delay, float rel_inh_strength, double poiss_rate):
-        ncells_exc_(nexc), ncells_inh_(ninh), ncells_ext_(next), delay_(delay), rate_(poiss_rate) {
+                  float weight, float delay, float rel_inh_strength, double poiss_rate, bool optimised):
+        ncells_exc_(nexc), ncells_inh_(ninh), ncells_ext_(next), delay_(delay), rate_(poiss_rate), optimised_(optimised) {
         // Make sure that in_degree_prop in the interval (0, 1]
         if (in_degree_prop <= 0.0 || in_degree_prop > 1.0) {
             std::out_of_range("The proportion of incoming connections should be in the interval (0, 1].");
         }
-        
+
         // Set up the parameters.
         weight_exc_ = weight;
         weight_inh_ = -rel_inh_strength * weight_exc_;
         weight_ext_ = weight;
         in_degree_exc_ = std::round(in_degree_prop * nexc);
         in_degree_inh_ = std::round(in_degree_prop * ninh);
-        in_degree_ext_ = std::round(in_degree_prop * ncells_ext_);
+        in_degree_ext_ = std::round(in_degree_prop * next);
     }
-    
+
     cell_size_type num_cells() const override {
-        return ncells_exc_ + ncells_inh_ + ncells_ext_;
+        if (optimised_) {
```

Comment:
Now that we have seen the benefits and validated the optimised version, let's remove the old version.